ios/brave-ios/Sources/AIChat/Subscription/SDK/AppStoreSDK.swift

```diff
@@ -0,0 +1,515 @@
+    let nonConsumable: [Product]
+    
+    /// Products the customer purchased that cannot be automatically renewed. Not a subscription product.
+    let nonRenewable: [Product]
+    
+    /// Products the customer purchased as a subscription that automatically renews. A subscription product.
+    let autoRenewable: [Product]
+    
+    /// All products the user has purchased or was given.
+    var all: [Product] {
+      return consumable + nonConsumable +
+             nonRenewable + autoRenewable
+    }
+    
+    init() {
+      self.consumable = []
+      self.nonConsumable = []
+      self.nonRenewable = []
+      self.autoRenewable = []
+    }
+    
+    init(consumable: [Product], nonConsumable: [Product], nonRenewable: [Product], autoRenewable: [Product]) {
+      self.consumable = consumable
+      self.nonConsumable = nonConsumable
+      self.nonRenewable = nonRenewable
+      self.autoRenewable = autoRenewable
+    }
+  }
+  
+  /// Returns all available productson the store
+  /// Each store must implement this function and return the list of products available
+  /// - Returns: A list of products available in the store
+  public var allAppStoreProducts: [any AppStoreProduct] {
+    fatalError("[AppStoreSDK] - Not Implemented")
+  }
+  
+  /// Products Available on the AppStore
+  @Published
+  private(set) var allProducts = Products()
+  
+  /// Products the customer has purchased
+  @Published
+  private(set) var purchasedProducts = Products()
+
+  // MARK: - Private
+  
+  /// Task that tracks AppStore product updates
+  private var updateTask: Task<Void, Error>?
+  
+  public init() {
+    // Start updater immediately
+    updateTask = Task.detached {
+      for await result in Transaction.updates {
+        do {
+          // Verify the transaction
+          let transaction = try self.verify(result)
+          
+          // Retrieve all products the user purchased
+          let purchasedProducts = await self.fetchPurchasedProducts()
+          
+          // Transactions must be marked as completed once processed
+          await transaction.finish()
+          
+          // Distribute the purchased products to the customer
+          await MainActor.run {
+            self.purchasedProducts = purchasedProducts
+          }
+        } catch {
+          Logger.module.error("[AppStoreSDK] - Transaction Verification Failed: \(error)")
+        }
+      }
+    }
+    
+    // Fetch initial products immediately
+    Task.detached {
+      // Retrieve all products the user purchased
+      await self.fetchProducts()
+      
+      // Retrieve all products the user purchased
+      let purchasedProducts = await self.fetchPurchasedProducts()
+      
+      // Distribute the purchased products to the customer
+      await MainActor.run {
+        self.purchasedProducts = purchasedProducts
+      }
+    }
+  }
+  
+  deinit {
+    // Stop listening to AppStore product updates
+    updateTask?.cancel()
```

Comment:
Keep in mind there is a SwiftUI bug in iOS 17.0-17.2 that doesn't deinit ObservableObject's that are injected into sheets, so you may want to move cleanup/cancellation to be explicit such as on a `onDisappear` modifier